pages/categories/index.js

```diff
@@ -41,35 +61,45 @@ export default function Categories({ shoppingItems, categories }) {
         <link rel="icon" href="/favicon.ico" />
       </Head>
 
-      <Header>Kategorien</Header>
+      {session ? (
+        <Header text="Kategorien">
+          <SignOutButton onSignOut={signOut} />
+        </Header>
+      ) : (
+        <Header text="MyShoppingManager" isOverlappingAnimation={true} />
+      )}
       <main>
-        <ContentWrapper>
-          <Background opacity="0.7" />
-          <CategoryContainer>
-            {filteredCategories.map(({ id, name, icon_src }) => (
-              <li key={id}>
-                <Link href={`/categories/${name}`}>
-                  <StyledButton>
-                    <span>{name}</span>
-                    <Image
-                      src={icon_src}
-                      width={30}
-                      height={30}
-                      alt={`${name} Icon`}
-                    />
-                  </StyledButton>
-                </Link>
-              </li>
-            ))}
-          </CategoryContainer>
-          <Link href={"/"} passHref>
-            <StyledLink>
-              <AllItemsButton />
-            </StyledLink>
-          </Link>
-        </ContentWrapper>
+        <Background opacity="0.7" />
+        {session && (
+          <ContentWrapper>
+            <Background opacity="0.7" />
+            <CategoryContainer>
+              {filteredCategories.map(({ id, name, icon_src }) => (
+                <li key={id}>
+                  <Link href={`/categories/${name}`}>
+                    <StyledButton>
+                      <span>{name}</span>
+                      <Image
+                        src={icon_src}
+                        width={30}
+                        height={30}
+                        alt={`${name} Icon`}
+                      />
+                    </StyledButton>
+                  </Link>
+                </li>
+              ))}
+            </CategoryContainer>
+            <Link href={"/"} passHref>
+              <StyledLink>
+                <AllItemsButton />
+              </StyledLink>
+            </Link>
+          </ContentWrapper>
+        )}
+        {!session && <SignIn onSignIn={signIn} />}
```

Comment:
I think you don't need the second conditional in line 100 if you change to
`session ? <ContentWrapper> ... </ContentWrapper> : <SignIn onSignIn={signIn} />`